hex/lib/dependabot/hex/file_parser.rb

```diff
@@ -137,6 +152,52 @@ def mixfiles
       def lockfile
         @lockfile ||= T.let(get_original_file("mix.lock"), T.nilable(Dependabot::DependencyFile))
       end
+
+      sig { returns(Ecosystem::VersionManager) }
+      def package_manager
+        @package_manager ||= T.let(
+          PackageManager.new(hex_version),
+          T.nilable(Dependabot::Hex::PackageManager)
+        )
+      end
+
+      sig { returns(T.nilable(Ecosystem::VersionManager)) }
+      def language
+        @language ||= T.let(
+          Language.new(elixir_version, language_requirement),
+          T.nilable(Dependabot::Hex::Language)
+        )
+      end
+
+      sig { returns(String) }
+      def hex_version
+        T.must(T.must(hex_info).fetch(:hex_version))
+      end
+
+      sig { returns(String) }
+      def elixir_version
+        T.must(T.must(hex_info).fetch(:elixir_version))
+      end
+
+      sig { returns(T.nilable(T::Hash[Symbol, T.nilable(String)])) }
+      def hex_info
+        @hex_info ||= T.let(begin
+          version = SharedHelpers.run_shell_command("mix hex.info")
+          {
+            hex_version: version.match(/Hex: \s*(\d+\.\d+(.\d+)*)/)&.captures&.first,
+            elixir_version: version.match(/Elixir: \s*(\d+\.\d+(.\d+)*)/)&.captures&.first
```

Comment:
seems to be a pattern, should this be a constant across ecosystems? Tagging @kbukum1 too

Reply:
I don't think so. It is ecosystem specific and even within the ecosystem the prefixes are different. One starts with hex, the package manager, and the other starts with elixir the language.